script/dfs.js

```diff
@@ -0,0 +1,77 @@
+var x,y; 
+
+function crossChecking(array) {
+    if ((array[0][0] == 1) || (array[array.length-1][array.length-1] == 1)) {
+        return "maze is impassable";
+    } else {
+        return "correct maze";
+    }
+}
+    
+function passWave(x, y, k, visited, maze) {
+    if (visited[x][y] == k-1) {
+        if ((x < maze.length-1) && (visited[x+1][y] == 0) && (maze[x+1][y] == 0)) {
+            visited[x+1][y] = k;
+        }
+        if ((x > 0) && (visited[x-1][y] == 0) && (maze[x-1][y] == 0)) {
+            visited[x-1][y] = k;
+        }
+        if ((y < maze.length-1) && (visited[x][y+1] == 0) && (maze[x][y+1] == 0)){
+            visited[x][y+1] = k;
+        }
+        if ((y > 0) && (visited[x][y-1] == 0) && (maze[x][y-1] == 0)) {
+            visited[x][y-1] = k;
+        }
+    }
+    return visited;
+}
+      
+function dfs(array, mazeArray) {
+    array[0][0] = 1;
```

Comment:
in general it is not recommended to work with array using indexes directly
because array can lost its optimizations